source/help/settings/channel-settings.rst

```diff
@@ -38,6 +38,14 @@ the channel topic or provide links to frequently accessed documents. Any
 channel member can edit this setting, unless the System Administrator
 has `restricted the permissions <https://docs.mattermost.com/administration/config-settings.html#enable-public-channel-renaming-for>`__.
 
+Adding links to the Channel Header
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Add links to the channel header using markdown.
+
```

Comment:
Did a quick test with bold, italics, emoji and code block and it seems all markdown is available in a channel header.